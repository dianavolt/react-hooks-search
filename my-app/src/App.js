import React, { useState } from 'react'
import ReactAutocomplete from 'react-autocomplete'
import {
    useSearch,
    useDebounce,
    useSearchForm
} from './hooks'

import Input from './components/Input'

function App() {
    const { searchVal, onSearchChange} = useSearchForm()
    const { articles } = useSearch(useDebounce(searchVal))

    return (
        <ReactAutocomplete
            items={articles}
            renderInput={Input}
            inputProps={{placeholder: 'Enter search term...'}}
            getItemValue={item => item.label}
            renderMenu={(children, value, style) => {
                return (articles && articles.length > 0)?
                    (<div style={{...style}} className="input-suggestions">
                        {children}
                        
                        <a href={`/search/query=${value}`} className='search-link'>
                            See all results
                        </a>
                    </div>
                    ) : <></>
            }}
            renderItem={(item, highlighted) =>
                <div
                    key={item.id}
                    style={{ backgroundColor: highlighted ? '#eee' : 'transparent'}}
                >
                    {item.label}
                </div>
            }
            value={searchVal}
            onChange={onSearchChange}
        />
    )
}

export default App