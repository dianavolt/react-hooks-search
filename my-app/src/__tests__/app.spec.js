import React from 'react'
import { mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'

import App from '../App'

// keyLength adds consistency to test snapshots
const render = (initialEntries) => <MemoryRouter keyLength={0} initialEntries={initialEntries}><App/></MemoryRouter>

describe('App component', () => {
    let test

    describe('when HOME is rendered', () => {
        beforeEach(() => {
            test = render(['/'])
        })

        it('should match the HOME snapshot', () => {
            expect(test).toMatchSnapshot()
        })
    })

    describe('when Not Found page is rendered', () => {
        beforeEach(() => {
            test = render(['/giberish'])
        })

        it('should match the Not Found snapshot', () => {
            expect(test).toMatchSnapshot()
        })
    })
})