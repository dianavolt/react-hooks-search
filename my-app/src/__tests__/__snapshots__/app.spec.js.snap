// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component when HOME is rendered should match the HOME snapshot 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      "/",
    ]
  }
  keyLength={0}
>
  <App />
</MemoryRouter>
`;

exports[`App component when Not Found page is rendered should match the Not Found snapshot 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      "/giberish",
    ]
  }
  keyLength={0}
>
  <App />
</MemoryRouter>
`;
